tsconfig.declaration.json

```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./tsconfig",
+  "compilerOptions": {
+    "noEmit": false,
+    "declaration": true,
+    "emitDeclarationOnly": true,
+    "allowJs": true
```

Comment:
add a check with `checkJS: true`, `yarn tsc` in another PR (non-breaking)

Reply:
Done in https://github.com/algolia/instantsearch.js/pull/4250